tests/test_special_ops.py

```diff
@@ -895,3 +893,37 @@ def test_accuracy_diag_embed(shape, dtype, offset, dim1, dim2):
     with flag_gems.use_gems():
         res_out = torch.diag_embed(inp, offset, dim1, dim2)
     gems_assert_equal(res_out, ref_out)
+
+
+def get_diagonal_backward_shape_and_dims():
+    shapes = SPECIAL_SHAPES
+    result = []
+
+    for s in shapes:
+        dim_pairs = get_dim1_dim2(len(s))
+        result.extend([(s, dim1, dim2) for dim1, dim2 in dim_pairs])
+
+    return result
+
+
+@pytest.mark.diagonal_backward
+@pytest.mark.parametrize("shape, dim1, dim2", get_diagonal_backward_shape_and_dims())
+@pytest.mark.parametrize("offset", [-1, 0, 1])
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
```

Comment:
are int dtypes supported?

Reply:
no, only float dtype tensor can calculate grad.
```python3
>>> import torch
>>> torch.ones(1, dtype=torch.int32, requires_grad=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Only Tensors of floating point and complex dtype can require gradients
```

Reply:
cases are different. leaf nodes can't require gradient.

Reply:
I don't quite understand. Could you please advise me on how to modify the unit test to support int dtypes?

Reply:
sorry I got it wrong.